src/views/Bookmark.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import BookmarkComponent from "../components/common/Bookmarks";
+import bookmarkIcon from "../assets/img/bookmark-icons.svg";
+import bookmarkedIcon from "../assets/img/bookmarked.svg";
+import { handleBookmark } from "../redux/actions/bookmarkActions";
+
+export class Bookmark extends Component {
+  render() {
+    const {
+      bookmark,
+      isBookmarked,
+      slug,
+      isSubmitting,
+      bookmarkedArticles
+    } = this.props;
+    let theBookmarkIcon;
+    if (!Object.prototype.hasOwnProperty.call(bookmarkedArticles, slug)) {
+      Object.assign(bookmarkedArticles, { [slug]: isBookmarked });
+    }
+    if (!bookmarkedArticles[slug]) {
+      theBookmarkIcon = bookmarkIcon;
+    } else {
+      theBookmarkIcon = bookmarkedIcon;
```

Comment:
I would suggest that you separate this `if else` from `render()` to avoid many lines of code per single method. You can create another component method to handle the checking you are doing here.

Reply:
Fixed